### PR TITLE
lzip: fix mingw regression

### PR DESCRIPTION
**Pull request by seanybaggins**
## Description of changes
fix mingw regression
https://github.com/NixOS/nixpkgs/pull/284905 broke the mingw build
relevant for https://github.com/NixOS/nixpkgs/issues/274274

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by seanybaggins** [comment]
@ofborg build pkgs.pkgsCross.mingw32.lzip

**Comment by seanybaggins** [comment]
@ofborg build pkgs.pkgsCross.mingwW64.lzip

**Comment by seanybaggins** [comment]
@wegank

**Comment by seanybaggins** [comment]
@ofborg build pkgs.pkgsCross.mingw32.lzip

**Comment by seanybaggins** [comment]
@ofborg build pkgs.pkgsCross.mingwW64.lzip

**Comment by Artturin** [comment]
This should be upstreamed

**Comment by seanybaggins** [comment]
@Artturin no extra work is needed on my end correct. You are just saying things should be upstreamed so another maintainer is in the loop?

**Comment by Artturin** [comment]
> @Artturin no extra work is needed on my end correct. You are just saying things should be upstreamed so another maintainer is in the loop?

"Upstream" in this context always refers to the source of the package, not nixpkgs

**Comment by seanybaggins** [comment]
I sent an email to lzip-bug@nongnu.org. Any chance we can merge these changes in the interim?

**Comment by wegank** [comment]
The regression seems already patched upstream: https://lists.nongnu.org/archive/html/lzip-bug/2024-02/msg00001.html

**Comment by wegank** [comment]
No, just pull the upstream fix (if possible).

**Comment by seanybaggins** [comment]
Sorry if this questioning is frustrating but I am still not 100% sure what I am supposed to be doing. Are you saying look for a patch file they have posted and do something like fetchpatch? Are you saying to create my own patch file that more closely follows the fix they made `#define mkdir(name,mode) _mkdir(name)`?

**Comment by seanybaggins** [comment]
Developer reached out to me. Asked me to test the release candidate.

**Comment by wegank** [comment]
> Sorry if this questioning is frustrating but I am still not 100% sure what I am supposed to be doing. Are you saying look for a patch file they have posted and do something like fetchpatch? Are you saying to create my own patch file that more closely follows the fix they made `#define mkdir(name,mode) _mkdir(name)`?

If upstream has a fix in their repo / mailing list, why not follow theirs. I'd be happy to merge the PR if the patch in https://lists.nongnu.org/archive/html/lzip-bug/2024-02/msg00015.html works.

File: pkgs/tools/compression/lzip/default.nix
Comment (bjornfor) [inline pkgs/tools/compression/lzip/default.nix:20]: Suggestion: Add a comment above the patch that says `Can be removed on next release:`. (IIUC)

Comment (seanybaggins) [reply pkgs/tools/compression/lzip/default.nix:23]: fixed

**Comment by seanybaggins** [comment]
> > Sorry if this questioning is frustrating but I am still not 100% sure what I am supposed to be doing. Are you saying look for a patch file they have posted and do something like fetchpatch? Are you saying to create my own patch file that more closely follows the fix they made `#define mkdir(name,mode) _mkdir(name)`?
> 
> If upstream has a fix in their repo / mailing list, why not follow theirs. I'd be happy to merge the PR if the patch in https://lists.nongnu.org/archive/html/lzip-bug/2024-02/msg00015.html works.
> 
> No need to fetchpatch, just replace the contents in `mingw-mkdir-one-argument.patch`.

Tried testing the new release candidates with the update and I still got the build error.

**Comment by wegank** [comment]
> Tried testing the new release candidates with the update and I still got the build error.

I only meant to test the patch in the message.

There are no new release candidates after 1.24. Check the dates here: http://download.savannah.gnu.org/releases/lzip/

**Review by bjornfor — APPROVED** [review]

**Pull request merged**
